dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/storage/S3Store.java

```diff
@@ -0,0 +1,98 @@
+package com.salesforce.dockerfileimageupdate.storage;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.salesforce.dockerfileimageupdate.utils.DockerfileGitHubUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class S3Store implements ImageTagStore {
+    private static final Logger log = LoggerFactory.getLogger(S3Store.class);
+    private static final Character S3_FILE_KEY_PATH_DELIMITER = ';';
+    public static final String s3Prefix = "s3";
+    private final AmazonS3 s3;
+    private final String store;
+
+    public S3Store(AmazonS3 s3, String store) {
+        this.s3 = s3;
+        this.store = store;
+    }
+
+    /* The store link should be a bucket name on S3. */
+    public void updateStore(String img, String tag) throws IOException {
+        if (store == null) {
+            log.info("Image tag store cannot be null. Skipping store update...");
+            return;
+        }
+        log.info("Updating store: {} with image: {} tag: {}...", store, img, tag);
+        if (s3.doesBucketExistV2(store)) {
+            String key = convertImageStringToS3ObjectKey(img);
+            s3.putObject(store, key, tag);
+        } else {
+            throw new IOException(String.format("The S3 bucket with name %s does not exist. Cannot proceed.", store));
```

Comment:
Java checked exceptions are stupid,  here is why http://literatejava.com/exceptions/checked-exceptions-javas-biggest-mistake/ .  In this case just throw and unchecked exception.  https://docs.oracle.com/javase/8/docs/api/java/io/UncheckedIOException.html is perfect for this case.

Also don't use string.format in Java.  It's literally the slowest formatting you can have, see https://redfin.engineering/java-string-concatenation-which-way-is-best-8f590a7d22a8

Reply:
For String.format, I'd rather have readability - I don't think its much of a perf hit all things considered. Good point though.

Reply:
I would agree on readability if there are a bunch of things being replaced.  But this is a single string value so

`"The S3 bucket with name " + store + " does not exist. Cannot proceed."`

is just as readable as

`String.format("The S3 bucket with name %s does not exist. Cannot proceed.", store)`

IMHO, Also it has fewer method calls and characters.